NanoAnalysis/python/genExtraFiller.py

```diff
@@ -0,0 +1,88 @@
+### 
+# Compute the gen-level angles. 
+#
+#
+###
+
+from __future__ import print_function
+from PhysicsTools.NanoAODTools.postprocessing.framework.eventloop import Module
+from PhysicsTools.NanoAODTools.postprocessing.framework.datamodel import Collection
+
+#from PhysicsTools.NanoAODTools.postprocessing.tools import deltaR
+from ctypes import c_float
+import Mela
+
+class genExtraFiller(Module):
+    def __init__(self, MELA):
+        print("***genExtraFiller")
+        self.MELA = MELA
+
+    def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+        self.out = wrappedOutputTree
+        # Book MELA angle branches
+        self.out.branch("GenZZ_qH", "F", limitedPrecision=12, title="The mass of the Higgs candidate as reconstructed by the 4-leptons at Gen-level.")
+        self.out.branch("GenZZ_mZ1", "F",  limitedPrecision=12, title="The mass of the first decay particle as reconstructed by 2 of the Gen-level leptons.")
+        self.out.branch("GenZZ_mZ2", "F",  limitedPrecision=12, title="The mass of the second decay particle as reconstructed by 2 of the Gen-level leptons.")
+        self.out.branch("GenZZ_costheta1", "F",  limitedPrecision=12, title="In the Higgs' rest frame, theta_1 is the angle between the momentum of Z1 and the momentum of one of its decay products.")
+        self.out.branch("GenZZ_costheta2", "F",  limitedPrecision=12, title="In the Higgs' rest frame, theta_2 is the angle between the momentum of Z2 and the momentum of one of its decay products.")
+        self.out.branch("GenZZ_Phi", "F", limitedPrecision=12, title="In the Higgs' rest frame, phi is the angle between the planes formed by the decay products of the two Z bosons.")
+        self.out.branch("GenZZ_costhetastar", "F", limitedPrecision=12, title="In the Higgs' rest frame, theta_star is the angle between the beamline and the momentum of one of the Higgs' decay products.")
+        self.out.branch("GenZZ_Phi1", "F",  limitedPrecision=12, title="In the Higgs' rest frame, phi_1 is the angle between the decay plane of Z1 and the beamline.")
+
+    def analyze(self, event) :
+        self.addExtra('GenPart', event)
+        
+
+        return True
+
+    def addExtra(self, collName, event) :
+        genpart = Collection(event, collName)
+        print("Z1l1idx: ", event.GenZZ_Z1l1Idx)
+       
+    
+        genLeps = [genpart[event.GenZZ_Z1l1Idx], genpart[event.GenZZ_Z1l2Idx], genpart[event.GenZZ_Z2l1Idx], genpart[event.GenZZ_Z2l2Idx]]
+
+ 
+        # MELA angle arrays
```

Comment:
I guess qH also needs to be initialized, otherwise there will be an error when MELA = None.
Also, line above: "angle arrays" -> "angles" 

Apart for this, all the rest looks fine.